R/preprocessing.R

```diff
@@ -0,0 +1,152 @@
+##' Fills missing data in a data set to prepare it for use within the package
+##'
+##' @description `r lifecycle::badge("experimental")`
+##'
+##' @param data Data frame with a `date` column. The other columns depend on the
+##'   model that the data are to be used, e.g. [estimate_infections()] or
+##'   [estimate_secondary()]. See the documentation there for the expected
+##'   format.
+##' @param missing Character. Options are "missing" (the default), "accumulate"
```

Comment:
When I first read this I didn't totally get what it was for but after a few passes I did. I wondered if a change name or a little sentence more clearly stating what missing was here would help but I couldn't come up with anything so perhaps it is just a difficult thing that people need to take the time to pass

Reply:
I think the thing that confused me most was `missing = "missing"`. Perhaps it should be `missing = "na"?

Reply:
ok, will try to rephrase / clarify in the function doc.

> I think the thing that confused me most was `missing = "missing"`. Perhaps it should be `missing = "na"`?

Hmm yes that does look weird - the problem is that you could have a pattern like `missing = "missing", na = "accumulate"` which I think would be even more confusing if then we had `missing = "na"`.

I'd suggest to:
- replace the choices with `c("ignore", "accumulate", "zero")` (with ignoring meaning ignore in the likelihood)
- rename the arguments from `missing` and `na` to `missing_dates` and `na_obs`, respectively

Reply:
> rename the arguments from missing and na to missing_dates and na_obs, respectively

I was coming to suggest this.